### PR TITLE
Add README & the tutorial to CI

### PR DESCRIPTION
**Pull request by Profpatsch**
Use @zimbatm’s mdsh tool to check both the readme and our tutorial on CI.

To prevent unwanted side-effects of the check commands, we use @aszlig’s `buildSandbox` utility when running `mdsh`, so that everything is done in a tmpfs.

The commits in this PR are quite diverse, the first one actually adds the ability to `lorri watch` to only do one build, I’m not sure whether the design is right (that’s why it’s still marked WIP).

Fixes https://github.com/target/lorri/issues/34
Fixes https://github.com/target/lorri/issues/47

A few improvements were made to `mdsh` and `buildSandbox` to support this workflow:
https://github.com/zimbatm/mdsh/pull/16
https://github.com/zimbatm/mdsh/pull/18
https://github.com/zimbatm/mdsh/pull/19
https://github.com/openlab-aux/vuizvui/pull/22
https://github.com/openlab-aux/vuizvui/pull/23

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -177,18 +229,25 @@ sub-directories are also watched for changes.
...
 
 Copyright 2019 Target
-License: Apache 2.0 (see [`LICENSE` file](./LICENSE))
 
+`$ head -n3 < ./LICENSE`
+```
+                                 Apache License
+                           Version 2.0, January 2004
+                        http://www.apache.org/licenses/
+```
```
Comment (grahamc) [inline README.md]: We should stick with the existing text here, as it was requested specifically by Target.

File: shell.nix [hunk 2]
```diff
@@ -1,6 +1,48 @@
 { pkgs ? import ./nix/nixpkgs.nix { enableMozillaOverlay = true; } }:
 let
   rustChannel = pkgs.latest.rustChannels.stable;
+
+  # command to run mdsh inside of a lightweight sandbox
+  # with a tmpfs root filesystem (that is deleted after execution).
```
Comment (grahamc) [inline shell.nix]: Let's move this definition to the overlay

File: shell.nix [hunk 3]
```diff
@@ -76,22 +126,45 @@ pkgs.mkShell rec {
       cargotestexit=$?
 
       travis_fold cargo-fmt \
-        sh -c 'cargo fmt && git diff --exit-code'
+        cargo fmt -- --check
```
Comment (grahamc) [inline shell.nix]: Nice.

**Comment by grahamc** [comment]
I'm thinking we should close this for now, since it hasn't received attention and the tests are failing. What do you think?

**Comment by Profpatsch** [comment]
It was only a small test failure related to sandboxing., I should really complete it.

File: nix/ci/sandbox.nix [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+          ([ [ mount "-t" "tmpfs" "container_root" newroot ] ]
+          # now mount the file system parts we need into the chroot
+          ++ builtins.concatMap
+               (rootPath: [
+                 [ "${pkgs.coreutils}/bin/mkdir" "-p" "${newroot}${rootPath}" ]
+                 [ mount "--rbind" rootPath "${newroot}${rootPath}" ]
+               ])
+               pathsToMount))
+        # finally, chroot into our new root directory
+        [ "${pkgs.coreutils}/bin/chroot" newroot "$@" ]
```
Comment (aszlig) [inline nix/ci/sandbox.nix]:
```suggestion
        [ "${pkgs.utillinux}/bin/unshare" "-U" "${pkgs.coreutils}/bin/chroot" newroot "$@" ]
```

**Comment by zimbatm** [comment]
I think one of the reason this PR hasn't been merged yet is that it's too big. Take a bit of time to split it into multiple PRs and it will be much easier.

**Comment by Profpatsch** [comment]
> I think one of the reason this PR hasn't been merged yet is that it's too big. Take a bit of time to split it into multiple PRs and it will be much easier.

yeah, I’ve been splitting out stuff all the while, and will probably split this into two separate PRs alltogether, one for the CI improvements and one for the mdsh part.

**Comment by gilligan** [comment]
Looks like there are various conflicts now and this hasn't been updated in half a year. Should this be closed or are you planning on returning to this @Profpatsch ?

**Comment by Profpatsch** [comment]
Realistically, this isn’t going anywhere, yes.

**Pull request closed without merging**